### DISCUSSION TITLE
Object reference not set to an instance of an object

### DISCUSSION THREAD

**Discussion by zsong** [dotnet/runtime#86390] [General]
The error message should be explicitly pointing out the **specific** object having the issue.

This can save many many hours of development work and make the world much better, especially make me healthier. This generic error message has bugged me for more than 15 years, and, still no efforts have been made to improve it.

Thanks.

**Comment by HaloFour** [comment]
This would be a [runtime](https://github.com/dotnet/runtime/) request as it's responsible for throwing the exception.  The runtime would have to interpret the IL instructions leading up to the exception in order to be able to format a description.

See: https://github.com/dotnet/runtime/issues/3858
